Agents/DistrictHeatingOptimisationAgent/README.md

```diff
@@ -0,0 +1,181 @@
+
+The agent is implemented as [derivation agent] using `DerivationWithTimeSeries`. Please note: 1) derivations with time series are currently restricted to synchronous derivations, i.e., derivations which get computed immediately upon request and 2) derivations with time series do not return any specific output triples, as all updates to the time series are expected to be conducted within the agent logic. Initial derivation output triples will only be generated when creating new derivation using `createSyncDerivationForNewInfo`.
+
+## 2.1 Generation Optimisation
+
+The implemented optimisation logic is described in detail in [preprint 275]. Upon successful optimisation, the following results are written back into the KG according to the [OntoHeatNet (Miro board)] ontology:
+- `ohn:ProvidedHeatAmount`: heat amount sourced from the EfW plant
+- `ohn:GeneratedHeatAmount`: heat amount provided by a certain heat generator, i.e., conventional gas boiler or co-gen gas turbine (instantiated for each heat generator)
+- `ohn:ConsumedGasAmount`: gas amount consumed by a certain heat generator to provide the required amount of heat (instantiated for each heat generator)
+- `ohn:CoGenElectricityAmount`: amount of co-generated electricty while providing required amount of heat (only relevant for gas turbine)
+- `ohn:Availability`: anticipated availability of a certain heat provider (instantiated for each heat generator and EfW plant)
+
+All optimisation outputs are instantiated as `ts:Forecast` instances for the respective concept (i.e., to not interfere with actual historical data instantiated via `om:hasValue` relationships). Newly created optimisation outputs automatically overwrite previously instantiated ones.
+
+Upon first invocation of the agent, historical gas consumption, heat generation, and (if applicable) electricity co-generation data is queried to fit generator specific gas consumption and co-gen models to be used during the optimisation. Those models will be reused for all subsequent optimisation requests.
+
+The agent is designed to handle each optimisation request individually, i.e., indipendent from any previous request. The only exception to this behaviour is when two subsequent requests are exactly +1h apart from one another. In this case, the second request will be considered dependent of the previous one to allow for MPC-style optimisation (i.e., keep track of gas turbine operation and previously accumulated benefit from current gas turbine activity).
+
+
+## 2.2 Optimisation Evaluation
+
+The agent also provides an HTTP endpoint to evaluate the total cost of the optimised vs. historical heat generation. It returns the cumulative generation cost over the entire optimisation interval as well as total and daily potential savings from the optimisation. It is triggered by receiving an empty HTTP `GET` request at the `/compare_cost` route. An example [cost comparison request] is provided.
+
+
+&nbsp;
+# 3. Using the Agent
```

Comment:
I would have hoped that a section titled "Using the agent" would contain a section on how to actually run it/make it do something. Perhaps move some things around from section 2? E.g. keep section 2 strictly about general background and move everything about running/triggering into section 3?

Reply:
Addressed by [b6d9c3c](https://github.com/cambridge-cares/TheWorldAvatar/pull/1031/commits/b6d9c3cf9df88d31bedcdff24afb653224a127c0)